lib/matrix_check_sum.rb

```diff
@@ -3,8 +3,33 @@
 # of numbers in row i is the same as the sum of numbers in column i for i = 0 to row.length-1
 # If this is the case, return true. Otherwise, return false.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2), n^2 are the total number of elements in the matrix
+# Space complexity: O(n), n is the size of the columns_sum array or the rows_sum_array
```

Comment:
You should be able to do this with `O(1)` space complexity. Think of how you might implement this without using an array for `columns_sum` and `rows_sum`.